src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7143,7 +7144,7 @@ public override void VisitForEachIterationVariables(BoundForEachStatement node)
                         }
 
                         // In non-error cases we'll only run this loop a single time. In error cases we'll set the nullability of the VariableType multiple times, but at least end up with something
-                        SetAnalyzedNullability(node.IterationVariableType, new VisitResult(result, destinationType), isLvalue: true);
+                        SetAnalyzedNullability(node.IterationVariableType, new VisitResult(isVar ? result : destinationType.ToTypeWithState(), destinationType), isLvalue: true);
```

Comment:
>isVar ? result : destinationType.ToTypeWithState() [](start = 91, length = 50)

It is not clear to me what is going on here. It appears we are adjusting (by comparison to what we used to do) an RValue type, but only for a non-var case. Why is this needed and why only for non-var case?